ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -792,6 +792,235 @@ int __process_config_file(const char *path, int *lno, int trust,
+		goto next_line;
+	}
+
+parse:
+	if (!off)
+		goto next_line;
+
+	if (ldmsd_is_initialized()) {
+		if ((0 == strncmp(line, "prdcr_add", 9)) ||
+				(0 == strncmp(line, "prdcr_start", 11))) {
+			if (strstr(line, "interval")) {
+				ldmsd_log(LDMSD_LWARNING,
+						"'interval' is being deprecated. "
+						"Please use 'reconnect' with 'prdcr_add' or 'prdcr_start*' "
+						"in the future.\n");
+			}
+		}
+	}
+
+	req_array = ldmsd_parse_config_str(line, msg_no, xprt.max_msg, ldmsd_log);
+	if (!req_array) {
+		rc = errno;
+		ldmsd_log(LDMSD_LERROR, "Process config string error at line %d "
+				"(%s). %s\n", lineno, cfg_str, STRERROR(rc));
+		goto cleanup;
+	}
+
+	request = __aggregate_records(req_array);
+	if (!request) {
+		rc = errno;
+		goto cleanup;
+	}
+	ldmsd_req_array_free(req_array);
+	req_array = NULL;
+
+	if (!ldmsd_is_initialized()) {
+		/* Process only the priority commands, e.g., cmd-line options */
+		if (!is_req_id_priority(ntohl(request->req_id)))
+			goto next_req;
+	} else {
+		/* Process non-priority commands, e.g., cfgobj config commands */
+		if (is_req_id_priority(ntohl(request->req_id)))
+			goto next_req;
+	}
+
+	/*
+	 * Make sure that LDMSD will create large enough buffer to receive
+	 * the config data.
+	 */
+	if (xprt.max_msg < ntohl(request->rec_len))
+		xprt.max_msg = ntohl(request->rec_len);
+
+	if (req_filter) {
+		rc = req_filter(&xprt, request, ctxt);
+		/* rc = 0, filter OK */
+		if (rc == 0) {
+			__dlog(DLOG_CFGOK, "# deferring line %d (%s): %s\n",
+				lineno, cfg_str, line);
+			goto next_req;
+		}
+		/* rc == errno */
+		if (rc > 0) {
+			ldmsd_log(LDMSD_LERROR,
+				  "Configuration error at "
+				  "line %d (%s)\n", lineno, cfg_str);
+			goto cleanup;
+		} else {
+			/* rc < 0, filter not applied */
+			rc = 0;
+		}
+	}
+
+	rc = ldmsd_process_config_request(&xprt, request);
+	if (rc || xprt.rsp_err) {
+		if (!rc)
+			rc = xprt.rsp_err;
+		ldmsd_log(LDMSD_LERROR, "Configuration error at line %d (%s)\n",
+				lineno, cfg_str);
+		goto cleanup;
+	}
+next_req:
+	free(request);
+	request = NULL;
+	msg_no += 1;
+	off = 0;
+	goto next_line;
+
+cleanup:
+	if (cfg_str)
+		free(cfg_str);
+	if (buff)
+		free(buff);
+	if (line)
+		free(line);
+	if (lno)
+		*lno = lineno;
+	ldmsd_req_array_free(req_array);
+	if (request)
+		free(request);
+	return rc;
+}
+
+char *__process_yaml_config_file(const char *path, const char *dname)
+{
+	FILE *fp;
+	char command[256];
+	char cstr[256];
+        char *cfg_str = malloc(256);
+	snprintf(command, sizeof(command), "ldmsd_yaml_parser --ldms_config %s --daemon_name %s 2>&1", path, dname);
+	fp = popen(command, "r");
+	if (!fp)
+		printf("Error in yaml_parser\n");
+	int lineno = 0;
+	size_t char_cnt = 0;
+	while (fgets(cstr, sizeof(cstr), fp) != NULL) {
+		printf("%s", cstr);
+		char_cnt += sizeof(cstr);
+		if (char_cnt >= 1024)
+			cfg_str = (char *)realloc(cfg_str, char_cnt - 256);
+		if (lineno)
+			strcat(cfg_str, cstr);
+		else
+			snprintf(cfg_str, sizeof(cstr), cstr);
```

Comment:
@nick-enoent this sizeof(cstr) is 8, not 256 like you intended. This is what is breaking the build.

Reply:
Addressed in the latest commit.